src/components/DataBrowser.vue

```diff
@@ -205,39 +205,37 @@ export default {
     },
     async fetchPaginatedFolderRows() {
       this.rowsLoading = true;
-      const { counts, location, options } = this;
+      const { counts: { nFolders, nItems }, location, options: { page, itemsPerPage } } = this;
       // if needed, get folder public info
       let folderNotPublic = false;
       if (!location.created && location._modelType === 'folder') {
         folderNotPublic = !(await this.girderRest.get(`folder/${location._id}`)).data.public;
       }
-      const { page, itemsPerPage } = options;
-      const folderParams = {
-        parentType: location._modelType,
-        parentId: location._id,
-        limit: itemsPerPage >= 0 ? itemsPerPage : null,
-        offset: (page - 1) * Math.abs(itemsPerPage),
-      };
-      const itemLimit = counts.nFolders > folderParams.offset
-        // if there are any folders on the current page,
-        // the numer of items to fetch is based on the number of folders
-        ? itemsPerPage - (counts.nFolders - folderParams.offset)
-        // else the page will be comprised of only items
-        : itemsPerPage;
-      const itemOffset = folderParams.offset - counts.nFolders;
-      const itemParams = {
-        folderId: location._id,
-        limit: itemsPerPage >= 0 ? itemLimit : null,
-        offset: itemOffset > 0 ? itemOffset : 0,
-      };
       const promises = [];
-      promises.push(this.girderRest.get(GIRDER_FOLDER_ENDPOINT, { params: folderParams }));
-      // a limit of < 0 signifies the current page only includes folders
-      // a limit of null signifies no options: fetch all entities
-      if (
-        (itemParams.limit > 0 || itemParams.limit === null) &&
-        location._modelType === 'folder'
-      ) {
+      const folderOffset = itemsPerPage !== -1 ? ((page - 1) * itemsPerPage) : 0;
+      const numberOfFoldersWillBeFetched = Math.max(0, nFolders - folderOffset);
+      if (numberOfFoldersWillBeFetched !== 0) {
+        const folderParams = {
+          parentType: location._modelType,
+          parentId: location._id,
+          limit: itemsPerPage !== -1 ? itemsPerPage : 0,
+          offset: folderOffset,
+        };
+        promises.push(this.girderRest.get(GIRDER_FOLDER_ENDPOINT, { params: folderParams }));
+      }
+      const itemOffset = folderOffset - nFolders > 0 ? folderOffset - nFolders : 0;
+      const numbersOfItemsWillBeFetched = nItems - itemOffset;
+      // if we could have items
+      if (location._modelType === 'folder'
+      // and there are items remaining
+      && numbersOfItemsWillBeFetched > 0
+      // and still have place for items
+      && (itemsPerPage === -1 || itemsPerPage - numberOfFoldersWillBeFetched > 0)) {
```

Comment:
This is an unusual style for multi-line conditionals.  Since airbnb is the Vue style guide we follow, https://github.com/airbnb/javascript/issues/1380 may be worth following.

https://github.com/airbnb/javascript#control-statements

Reply:
Sounds good.